aspose/src/main/java/nl/nn/adapterframework/extensions/aspose/pipe/PdfPipe.java

```diff
@@ -164,6 +173,21 @@ public PipeRunResult doPipe(Message input, PipeLineSession session) throws PipeR
 		}
 	}
 
+	private void populateContext(CisConversionResult result, Message message, int index) {
+		if(StringUtils.isNotEmpty(result.getResultFilePath())){
+			FileMessage document = new FileMessage(new File(result.getResultFilePath()));
+			message.getContext().put(CONVERTED_DOCUMENTS_CONTEXT_KEY+index, document);
+		}
+
+		List<CisConversionResult> attachmentList = result.getAttachments();
+		if (attachmentList != null && !attachmentList.isEmpty()) {
+			for (int i = 0; i < attachmentList.size(); i++) {
+				index++;
+				populateContext(attachmentList.get(i), message, index);
```

Comment:
@Laurens-makel Did you test this recursion? It looks to me like this will not correctly update the index for all nested documents.

Reply:
I've done some debugging both in my branch and in your branch and I never see an attachment that has a file that is written out.
That makes me concerned whether we are doing this right, or if the Aspose module is actually broken and we never noticed because our tests are too dumb.

Reply:
Update: I now see that we have an option to save separate PDFs but that currently we never test this. 🙄 
That explains a lot.